test/test_parser.cpp

```diff
@@ -6405,4 +6405,69 @@ TEST(Parser_issue6831)
     CHECK_EQUAL(q.count(), 1);
 }
 
+TEST(Parser_issue7393)
+{
+    Group g;
+    auto table = g.add_table_with_primary_key("Table", type_String, "id");
+    auto col_value = table->add_column(type_Mixed, "value");
+    auto obj = table->create_object_with_primary_key("EMBEDDED");
+    obj.set_json(col_value, R"([
+        [4, 5, 6],
+        {
+            "key1": 7,
+            "key2": 8,
+            "key3": [9]
+        }
+    ])");
+    auto q = table->query("value[*][*] == {4, 5, 6}");
+    CHECK_EQUAL(q.count(), 1);
+    q = table->query("value[0][*] == {4, 5, 6}");
+    CHECK_EQUAL(q.count(), 1);
+    q = table->query("value[1][*] == {4, 5, 6}");
+    CHECK_EQUAL(q.count(), 0);
+}
+
+TEST(Parser_Wildcard)
+{
+    Group g;
+    auto table = g.add_table_with_primary_key("table", type_String, "id");
+    auto origin = g.add_table_with_primary_key("origin", type_String, "id");
+    auto col_any = table->add_column(type_Mixed, "value");
+    auto col_dict = table->add_column_dictionary(type_Mixed, "dict");
+    auto col_list = table->add_column_list(type_Mixed, "list");
+    auto col_link = origin->add_column_list(*table, "link");
+
+    auto obj1 = table->create_object_with_primary_key("obj1");
+    obj1.set_json(col_any, R"([[3, 2, 1], [4, 5, 6]])");
+    obj1.set_json(col_dict, R"({"key1": 1, "key2": 2, "key3": [4, 5, 6]})");
+    obj1.set_json(col_list, R"([4, 5, 6])");
+    auto obj2 = table->create_object_with_primary_key("obj2");
+    obj2.set_json(col_any, R"([[9, 8, 7], [1, 2, 3]])");
+    obj2.set_json(col_dict, R"({"key1": 3, "key2": 4, "key3": [3, 2, 1]})");
+    obj2.set_json(col_list, R"([1, 2, 3])");
+
+    auto obj = origin->create_object_with_primary_key("top");
+    auto ll = obj.get_linklist(col_link);
+    ll.add(obj1.get_key());
+    ll.add(obj2.get_key());
+
+    auto q = table->query("value[*][*] = {1, 2, 3, 4, 5, 6}");
+    CHECK_EQUAL(q.count(), 0);
+    q = table->query("value[*][*] = {3, 2, 1}");
+    CHECK_EQUAL(q.count(), 1);
+    q = table->query("{3, 2, 1} = value[*][*]");
+    CHECK_EQUAL(q.count(), 1);
+    q = table->query("value[*][*] = dict[*][*]");
```

Comment:
Please make sure that we also have meaningful comparisons when there is a expression comparison type involved as well. When comparing many-to-many lists, my intuition is that it is correct to apply ANY/ALL/NONE to the final list, not the amalgamation of all lists combined. I think this is how the code works as you have it written. But to be sure can you try adding a few combinations? eg:
```
ANY value[*][*] = ANY dict[*][*] // eg: {{1, 2, 3}, {4, 5, 6}} should match on {{1}, {10}} but not on {{4}, {1}}
ALL value[*][*] <= ANY dict[*][*] // eg: {{1, 2, 3}, {4, 5, 6}} should match on {{1, 10}, {10}} but not on {{2, 10}, {10}}
ALL value[*][*] = ALL dict[*][*]
NONE value[*][*] = ANY dict[*][*]
```

Reply:
I added a few more tests.